functions/Set-DbaSqlLog.ps1

```diff
@@ -0,0 +1,162 @@
+
+        .EXAMPLE
+            Set-DbaSqlLogConfig -SqlInstance server2012 -NumberOfLog 25 -SizeInKb 500
+
+            Sets the number of error log files to 25 and size before it will roll over to 500KB on server2012
+    #>
+    [cmdletbinding(SupportsShouldProcess)]
+    param(
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [DbaValidatePattern('^[6-9][0-9]?$', ErrorMessage = "Error processing {0} - input must be an integer between 6 and 99")]
+        [int]$NumberOfLog,
+        [int]$SizeInKb,
+        [switch]$EnableException
+    )
+    process {
+        foreach ($instance in $SqlInstance) {
+            Write-Message -Level Verbose -Message "Connecting to $instance"
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            $collectionNumberLog = [PSCustomObject]@{
+                ComputerName            = $server.NetName
+                InstanceName            = $server.ServiceName
+                SqlInstance             = $server.DomainInstanceName
+                Setting                 = 'NumberOfLogFiles'
+                OriginalNumberErrorLogs = $null
+                CurrentNumberErrorLogs  = $null
+                Status                  = $null
+            }
+            $collectionErrorLogSize = [PSCustomObject]@{
+                ComputerName           = $server.NetName
+                InstanceName           = $server.ServiceName
+                SqlInstance            = $server.DomainInstanceName
+                Setting                = 'ErrorLogSizeKb'
+                OriginalErrorLogSizeKb = $null
+                CurrentErrorLogSizeKb  = $null
+                Status                 = $null
+            }
+            if (Test-Bound 'NumberOfLog') {
+                try {
+                    $currentNumLogs = $server.NumberOfLogFiles
+                    $collectionNumberLog.OriginalNumberErrorLogs = $currentNumLogs
+                }
+                catch {
+                    $collectionNumberLog.Status = "Failed collection"
+                    $collectionNumberLog
+                    Stop-Function -Message "Issue collecting current value for number of error logs" -Target $server -ErrorRecord $_ -Exception $_.Exception.InnerException.InnerException.InnerException -Continue
+                }
+
+                if ($NumberOfLog -eq $currentNumLogs) {
+                    if ($PSCmdlet.ShouldProcess($server, "Provide warning that NumberOfLog value already matches the configured value")) {
+                        Write-Message -Level Warning -Message "The provided value for NumberOfLog is already set" -Continue
+                    }
```

Comment:
Looks good in `-WhatIf`, but makes little sense in Production use (asking for a confirm to do nothing)
Instead could use `Test-Bound 'WhatIf'` to check for WhatIf use and change messages accordingly